File: src/pocketcloud/server/data/CloudServerData.php
<?php

namespace pocketcloud\server\data;

class CloudServerData {

    public function __construct(
        private readonly int $port,
        private int $maxPlayers,
        private ?int $processId = null
    ) {}

    public function setMaxPlayers(int $maxPlayers): void {
        $this->maxPlayers = $maxPlayers;
    }

    public function setProcessId(?int $processId): void {
        $this->processId = $processId;
    }

    public function getPort(): int {
        return $this->port;
    }

    public function getMaxPlayers(): int {
        return $this->maxPlayers;
    }

    public function getProcessId(): ?int {
        return $this->processId;
    }
}

File: cloud-shared/src/main/java/de/pocketcloud/shared/component/data/CloudServerData.java
package de.pocketcloud.shared.component.data;

import de.pocketcloud.api.server.data.ICloudServerData;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.UUID;

@Getter
@Setter
@Accessors(fluent = true)
@AllArgsConstructor
public class CloudServerData implements ICloudServerData {

    protected final UUID serverUuid;
    protected final String address;
    protected final int port;
    protected int maxPlayers;

    protected Long processId = null;
    protected Double tps = null;
    protected Double avgTps = null;
    protected Long memoryUsage = null;
    protected Long memoryPeak = null;
    protected Long memoryLimit = null;
    protected Double cpuUsage = null;

    public CloudServerData(UUID serverUuid, String address, int port, int maxPlayers) {
        this.serverUuid = serverUuid;
        this.address = address;
        this.port = port;
        this.maxPlayers = maxPlayers;
    }

    public void setPerformanceStats(Double tps, Double avgTps, Long memoryUsage, Long memoryPeak, Long memoryLimit, Double cpuUsage) {
        this.tps = tps;
        this.avgTps = avgTps;
        this.memoryUsage = memoryUsage;
        this.memoryPeak = memoryPeak;
        this.memoryLimit = memoryLimit;
        this.cpuUsage = cpuUsage;
    }
}